exp/cloud/services/scalesets/vmss.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scalesets
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-12-01/compute"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/pkg/errors"
+	"k8s.io/klog"
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	azure "sigs.k8s.io/cluster-api-provider-azure/cloud"
+	"sigs.k8s.io/cluster-api-provider-azure/cloud/converters"
+	convertersexp "sigs.k8s.io/cluster-api-provider-azure/exp/cloud/converters"
+)
+
+// Spec contains properties to create a managed cluster.
+// Spec input specification for Get/CreateOrUpdate/Delete calls
+type (
+	Spec struct {
+		Name       string
+		Sku        string
+		Capacity   int64
+		SSHKeyData string
+		Image      *infrav1.Image
+		OSDisk     infrav1.OSDisk
+		CustomData string
+	}
+)
+
+func (s *Service) Get(ctx context.Context, spec interface{}) (interface{}, error) {
```

Comment:
Consider deprecating this `spec interface{}` pattern. It was added before with the promise that it would be fixed in a subsequent PR.

Reply:
Would love to! 

We have discussed this a few times and intend on providing a refactoring proposal. It would be great to get your input on that proposal.